### PR TITLE
[Work in progress] add support for sending empty payload frame (gateway issue #254)

### PR DESCRIPTION
**Pull request by danibusu**
Changed just the write() method of the AbstractIoSession class in the mina.netty module to not throw IllegalArgumentException, as debugging revelead that this class is the one being used, and not the one in the mina.core module.
Should I also perform the change in the AbstractIoSession from mina.core? When is one class used over the other? Also, what's the purpose of the mina.build module?
I removed the @Ignore annotations from impacted ITs and they all pass.

### CODE REVIEW

**Comment by cmebarrow** [comment]
@danibusu I think we only use the AbstractIoSession from mina-netty.

File: mina.netty/src/main/java/org/kaazing/mina/core/session/AbstractIoSession.java [hunk 1]
```diff
@@ -99,6 +101,8 @@ public void operationComplete(CloseFuture future) {
             }
     };
 
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractIoSession.class);
```
Comment (jitsni) [inline mina.netty/src/main/java/org/kaazing/mina/core/session/AbstractIoSession.java]: May be s/LOG/LOGGER

File: mina.netty/src/main/java/org/kaazing/mina/core/session/AbstractIoSession.java [hunk 2]
```diff
@@ -426,9 +430,9 @@ public WriteFuture write(Object message, SocketAddress remoteAddress) {
         try {
```
Comment (jitsni) [inline mina.netty/src/main/java/org/kaazing/mina/core/session/AbstractIoSession.java]: It is there to make sure to catch application errors. I guess we can remove the while try { } block.

But if possible we should limit this change to websocket transport only. Can you paste the complete stack trace  ?

**Comment by danibusu** [comment]
@jitsni I pasted the stack trace below:

<pre><code>
Thread [main] (Suspended (breakpoint at line 430 in AbstractIoSession))	
	WsnSession(AbstractIoSession).write(Object, SocketAddress) line: 430	
	WsnSession(AbstractIoSession).write(Object) line: 391	
	BaseFramingIT.shouldEchoBinaryFrameWithPayloadLength0() line: 97	
	NativeMethodAccessorImpl.invoke0(Method, Object, Object[]) line: not available [native method]	
	NativeMethodAccessorImpl.invoke(Object, Object[]) line: 62	
	DelegatingMethodAccessorImpl.invoke(Object, Object[]) line: 43	
	Method.invoke(Object, Object...) line: 498	
	FrameworkMethod$1.runReflectiveCall() line: 50	
	FrameworkMethod$1(ReflectiveCallable).run() line: 12	
	FrameworkMethod.invokeExplosively(Object, Object...) line: 47	
	InvokeMethod.evaluate() line: 17	
	JUnitRuleMockery$1.evaluate() line: 49	
	SpecificationStatement.evaluate() line: 63	
	Verifier$1.evaluate() line: 35	
	WsnConnectorRule$ConnectorStatement.evaluate() line: 134	
	TestWatcher$1.evaluate() line: 55	
	RunRules.evaluate() line: 20	
	BlockJUnit4ClassRunner(ParentRunner<T>).runLeaf(Statement, Description, RunNotifier) line: 325	
	BlockJUnit4ClassRunner.runChild(FrameworkMethod, RunNotifier) line: 78	
	BlockJUnit4ClassRunner.runChild(Object, RunNotifier) line: 57	
	ParentRunner$3.run() line: 290	
	ParentRunner$1.schedule(Runnable) line: 71	
	BlockJUnit4ClassRunner(ParentRunner<T>).runChildren(RunNotifier) line: 288	
	ParentRunner<T>.access$000(ParentRunner, RunNotifier) line: 58	
	ParentRunner$2.evaluate() line: 268	
	BlockJUnit4ClassRunner(ParentRunner<T>).run(RunNotifier) line: 363	
	JUnit4TestReference.run(TestExecution) line: 86	
	TestExecution.run(ITestReference[]) line: 38	
	RemoteTestRunner.runTests(String[], String, TestExecution) line: 459	
	RemoteTestRunner.runTests(TestExecution) line: 675	
	RemoteTestRunner.run() line: 382	
	RemoteTestRunner.main(String[]) line: 192	
</pre></code>

I am thinking of updating the WsnConnector and WsnAcceptor classes and whenever a new WsnSession is created, also set a session attribute like "wsFlag" to true. And then, in the AbstractIoSession.write() method, throw the IllegalArgumentException only if the flag is not set. Do you think this approach is the right one?

I'm not sure about removing the try-catch block. For example, the FileInputStream constructor called in one of the if  blocks may throw FileNotFoundException.

**Comment by jitsni** [comment]
One option is to create a method void validateWriteMessage((Object message) with default behaviour.  WsnSession overrides this method to allow empty message. But seems like heavy weight.

I think the current change is ok. Just remove the whole try block. I don't we use File or FileChannel as message.

**Comment by danibusu** [comment]
@jitsni Updated pull request to incorporate review comments. Thanks!

**Pull request merged**